tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/GraphQL/HotChocolate/IQueryResult.cs

```diff
@@ -0,0 +1,22 @@
+// <copyright file="IQueryResult.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.GraphQL.HotChocolate
+{
+    /// <summary>
+    /// HotChocolate.Execution.IQueryResult interface for ducktyping
+    /// https://github.com/ChilliCream/graphql-platform/blob/35301472065248ce4e2f34894041f39124e3c7b8/src/HotChocolate/Core/src/Abstractions/Execution/IQueryResult.cs
+    /// </summary>
+    internal interface IQueryResult
+    {
+        /// <summary>
+        /// Gets the executing operation errors
+        /// </summary>
+        IEnumerable Errors { get; }
+    }
```

Comment:
I _think_ this should be a struct? We don't use it in an integration type constraints, and we don't need to call any methods on it, so we should favor a struct proxy in this case. See [this doc](https://github.com/DataDog/dd-trace-dotnet/blob/master/docs/development/DuckTyping.md#best-practices) for more info 🙂

Reply:
The `IQueryResult.cs` is used in the `ExecuteAsyncIntegrationV13.cs`. It's used to cast an `IExecutionResult` result to a `IQueryResult`.
I tried to change the `IQueryResult.cs` to a struct but the cast won't work anymore after that.

Reply:
Oh yeah, so it is, sorry 👍 Disregard 😄

This is an edge case where we should really have an extra helper method so that you _can_ use a struct, but that's outside of the scope of things here!